Modules/Exercise/Assignment/class.ilExAssignment.php

```diff
@@ -980,13 +1022,43 @@ public static function cloneAssignmentsOfExercise(
 
     public function getFiles(): array
     {
-        $this->log->debug("getting files from class.ilExAssignment using ilFSWebStorageExercise");
-        $storage = new ilFSWebStorageExercise($this->getExerciseId(), $this->getId());
-        return $storage->getFiles();
+        if ($this->hasInstructionFileRCID()) {
+            $collection = $this->irss->collection()->get($this->getInstructionFileRCID());
+            return array_map(function (ResourceIdentification $rid): array {
+                // this return the same array as the ilFSWebStorageExercise implementation for compatibility
+                $info = $this->irss->manage()->getResource($rid)
+                    ->getCurrentRevision()
+                    ->getInformation();
+                // if possible we use a flavour (PreviewDefinition from the ilResourceCollectionGUI)
+                $flavour_definition = new PreviewDefinition();
```

Comment:
@chfsx Why is the PreviewDefinition flavour used here. I did not find anything about it in a README.md. Why not using the original files?

Reply:
you could use the original files as well, but those files can be huge in size. using a Flavour reduces the size of images to a reasonable size. but I can remove this if you do not want it...